### DISCUSSION TITLE
Ring MQTT Can't Authenticate Since Ring Outage

### DISCUSSION THREAD

**Discussion by jamesphw** [tsightler/ring-mqtt#694] [Support Q&A] [answered]
As most of you are probably aware, Ring had a major global outage last week.

Since the outage, I haven't been able to get ring-mqtt to successfully re-authenticate. I can enter my credentials, the token appears to be renewed, but then it complains that it's not a valid token.  This is all I see in the logs when I generate a new token...

`2023-07-30T19:56:21.664Z ring-mqtt Username/Password was accepted, waiting for 2FA code to be entered.`
`2023-07-30T19:56:30.472Z ring-mqtt Saving updated refresh token to state file`
`2023-07-30T19:56:30.473Z ring-mqtt A new refresh token was generated, attempting to re-establish connection to Ring API`
`2023-07-30T19:56:31.506Z ring-mqtt Successfully saved updated state file: /data/ring-state.json`
`2023-07-30T19:56:32.477Z ring-mqtt Refresh token is not valid.  Unable to authenticate with Ring servers.  See https://github.com/dgreif/ring/wiki/Refresh-Tokens`
`2023-07-30T19:56:32.478Z ring-mqtt Failed to re-establish connection to Ring API using generated refresh token`

Anyone know if this is just an ongoing unfixed issue, or if there is something else that needs to be done to fix this?

**Comment by tsightler** [comment, marked as answer]
Have you followed the standard process of stopping the addon, removing the equivalent device from Ring Control Center, and then re-start the addon and re-authenticate?

Also, have you actually restarted the addon (not just an HA restart, which doesn't restart addons).  I'm always suspect when people post only log snippets that they haven't actually restarted and re-authenticated from that point.  There are some cases where re-authenticating when it was already running just doesn't re-initialize the connection properly, usually because something like the websocket is using the old token but new polling sessions don't work.

I didn't need to re-authenticate at all for my instances, just waited for the outage to pass and restarted the addon, everything worked fine.  I'd guess that 99% of issues are resolved by an addon restart, but almost nobody seems to do it (most do an HA restart which they think restarts addons, even though it doesn't).

**Comment by jamesphw** [comment]
Thanks for the answer @tsightler ! Restarting the add-on fixed the problem. Indeed I was restarting HA, and I had no idea that it doesn't also restart the add-on.

**Reply by tsightler** [reply]
Yep choosing "Restart Home Assistant" only restarts the container running HA, not the other containers running addons.  I wish this was more clear in the HA UI.  You can restart everything by opening "Advanced options" and selecting "Reboot system" which will actually shutdown the OS and restart everything.

I've put this in the FAQ but maybe I should add some notes in the addon Web UI or maybe force the addon to restart when users re-authenticate.
